src/ValidatingObserver.php

```diff
@@ -83,10 +83,63 @@ protected function validate(Model $model)
 
         $messages = isset($model->validationMessages) ? $model->validationMessages : [];
 
-        $validator = $this->factory->make($attributes, $model->rules, $messages);
+        $validator = $this->factory->make($attributes, $this->getRules($model), $messages);
 
         if ($validator->fails()) {
             throw new ValidationException($validator->getMessageBag());
         }
     }
+
+    /**
+     * Get the model validation rules.
+     *
+     * @param \Illuminate\Database\Eloquent\Model $model
+     *
+     * @return array
+     */
+    public function getRules($model)
+    {
+        $rules = $model->rules;
+
+        if (!$model->exists) {
+            return $rules;
+        }
+
+        foreach ($rules as $field => $ruleset) {
+            $ruleset = is_string($ruleset) ? explode('|', $ruleset) : $ruleset;
+
+            foreach ($ruleset as $rule) {
+                if (starts_with($rule, 'unique')) {
+                    $rules[$field] = $this->addExceptId($model, $rule);
+                }
+            }
+        }
+
+        return $rules;
+    }
+
+    /**
+     * Add except id to unique rules.
+     *
+     * @param \Illuminate\Database\Eloquent\Model $model
+     * @param string                              $rule
+     *
+     * @return string
+     */
+    public function addExceptId($model, $rule)
```

Comment:
The unique rule is `unique:table,column,except,idColumn` so when you are validating the model on update having the rule it would fail always because you need to specify if the record it's already on the database. You do that by appending the id to except and the name of the id column. So what we are doing here is to check if the model is set for updating and if it is we append the model key and the key name to the rule.